prow/plank/controller.go

```diff
@@ -359,7 +361,7 @@ func (c *Controller) syncPendingJob(pj kube.ProwJob, pm map[string]kube.Pod, rep
 			pj.Status.Description = "Job succeeded."
 			for _, nj := range pj.Spec.RunAfterSuccess {
 				child := pjutil.NewProwJob(nj, pj.ObjectMeta.Labels)
-				if !c.RunAfterSuccessCanRun(&pj, &child, c.ca, c.ghc) {
+				if c.ghc != nil && !c.RunAfterSuccessCanRun(&pj, &child, c.ca, c.ghc) {
```

Comment:
I didn't know that Plank was using a Github client for anything besides reporting. If we want to completely remove Plank's coupling with Github we need to resolve this in addition to https://github.com/kubernetes/test-infra/issues/7741.
I think it could be worth banning run_after_success jobs from using run_if_changed in order to get rid of this dependency.
@kargakis What do you think?

Reply:
> I think it could be worth banning run_after_success jobs from using run_if_changed in order to get rid of this dependency.

Hrm, I don't feel strong about scoping down the api for this. We don't use it today but it's a fine thing to support. Is there a proposal for adding gerrit support that I can read? Can we say that we don't support run_after_success + run_if_change only for gerrit jobs (assuming there is no way to grab changes from gerrit)? How difficult+expensive is fetching a patch from gerrit and figuring out its changeset?

Reply:
hummm, the design doc is internal so I cannot share that yet

but the controller is already checked in and deployed, you can reference https://github.com/kubernetes/test-infra/tree/master/prow/cmd/gerrit and https://github.com/kubernetes/test-infra/tree/master/prow/gerrit

Reply:
> We don't use it today but it's a fine thing to support.

By this I meant we ar enot using it in Openshift. I think it's used in k8s by kubeadm.